middleman_protocol/tests/test_helpers.py

```diff
@@ -0,0 +1,353 @@
+        assertpy.assert_that(deserialized_message.payload).is_instance_of(type(payload))
+        assertpy.assert_that(deserialized_message.payload).is_equal_to(payload)
+
+    def test_that_receiving_a_series_of_messages_should_be_handled_correctly(self):
+        payload = Ping()
+        middleman_message = GolemMessageFrame(payload, self.request_id)
+
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as server_socket:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+                server_socket.bind(('127.0.0.1', 8001))
+                server_socket.listen(1)
+
+                client_socket.connect(('127.0.0.1', 8001))
+
+                (connection, _address) = server_socket.accept()
+
+                for _i in range(10):
+                    send_over_stream(
+                        connection=client_socket,
+                        raw_message=middleman_message,
+                        private_key=CONCENT_PRIVATE_KEY
+                    )
+
+                for _i in range(10):
+                    deserialized_message = next(receive_frame(connection=connection, public_key=CONCENT_PUBLIC_KEY))
+
+                    assertpy.assert_that(deserialized_message).is_instance_of(GolemMessageFrame)
+                    assertpy.assert_that(deserialized_message.payload).is_instance_of(Ping)
+                    assertpy.assert_that(deserialized_message.payload).is_equal_to(payload)
+
+    def test_that_receiving_wrongly_encoded_message_should_return_none(self):
+        middleman_message = GolemMessageFrame(Ping(), self.request_id)
+        raw_message = middleman_message.serialize(
+            private_key=CONCENT_PRIVATE_KEY,
+        )
+
+        raw_message_encoded = escape_encode_raw_message(raw_message)
+        raw_message_encoded = raw_message_encoded + ESCAPE_CHARACTER + b'\xff'
+        raw_message_encoded = append_frame_separator(raw_message_encoded)
+
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as server_socket:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+            with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+                server_socket.bind(('127.0.0.1', 8001))
+                server_socket.listen(1)
+
+                client_socket.connect(('127.0.0.1', 8001))
+
+                (connection, _address) = server_socket.accept()
+
+                client_socket.send(raw_message_encoded)
+                raw_message_received = next(receive_frame(connection=connection, public_key=CONCENT_PUBLIC_KEY))
+
+        assertpy.assert_that(raw_message_received).is_equal_to(None)
```

Comment:
for `None` comparison, please use: `assertpy.assert_that(raw_message_received).is_none()`